### ISSUE TITLE
Search

### ISSUE THREAD

**Issue by not-matthias** [not-matthias/apollo#35]
Should use WASM-based libraries to reduce the network footprint:
- https://github.com/cloudcannon/pagefind
- https://github.com/tinysearch/tinysearch (only support word search)

**Comment by SeniorMars** [comment]
Counterpoint: [elasticlunr](http://elasticlunr.com/) is built in and using pagefind/tinysearch will make it hard to use with github actions / require you to self host.

**Comment by SeniorMars** [comment]
Do you want me to implement this with [elasticlunr](http://elasticlunr.com/)? I think it is a good idea for a default. We can disable it by default too.

**Comment by not-matthias** [comment]
Yeah, sounds good. Don't think it'll make a big difference if it's wasm or js. Feel free to use elasticlunr. And keeping it turned off by default is also a good idea. :+1:

**Comment by soumendrak** [comment]
@SeniorMars / @not-matthias 
Is there any progress on this feature? Or can you please guide me which all files do I need to add/update for this search functionality?

**Comment by epicwhale** [comment]
this tabi theme (https://github.com/welpo/tabi?tab=readme-ov-file) has a reference elasticlunr implementation if it helps!
